From: Nathan Pratta Teodosio <nathan.teodosio@canonical.com>
Date: Mon, 12 Aug 2024 20:35:16 +0200
Subject: fix_FTBFS_s390x
Forwarded: not-yet
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

Don't treat promotion from float to double as error in s390x.
 .
 Specifically on s390x, the compiler gives a bunch of errors such as
 .
 /<<PKGBUILDDIR>>/intern/cycles/kernel/../util/math.h:476:9: error: implicit conversion from ‘float’ to ‘double’ to match other operand of binary expression [-Werror=double-promotion]
   476 |   if (f == 0.0f) {
       |       ~~^~~~~~~
 .
 where it's easy to see that all the operands are actually floats.
---
 intern/cycles/kernel/CMakeLists.txt | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/intern/cycles/kernel/CMakeLists.txt b/intern/cycles/kernel/CMakeLists.txt
index 814da50..95f045f 100644
--- a/intern/cycles/kernel/CMakeLists.txt
+++ b/intern/cycles/kernel/CMakeLists.txt
@@ -1349,11 +1349,10 @@ endif()
 
 # Warnings to avoid using doubles in the kernel.
 if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_C_COMPILER_ID MATCHES "Clang")
-  add_check_cxx_compiler_flags(
-    CMAKE_CXX_FLAGS
-    _has_cxxflag_float_conversion "-Werror=float-conversion"
-    _has_cxxflag_double_promotion "-Werror=double-promotion"
-  )
+  add_check_cxx_compiler_flags(CMAKE_CXX_FLAGS _has_cxxflag_float_conversion "-Werror=float-conversion")
+  if(NOT CMAKE_SYSTEM_PROCESSOR MATCHES "s390x")
+    add_check_cxx_compiler_flags(CMAKE_CXX_FLAGS _has_cxxflag_double_promotion "-Werror=double-promotion")
+  endif()
   unset(_has_cxxflag_float_conversion)
   unset(_has_cxxflag_double_promotion)
 endif()
